sycl/test/basic_tests/SYCL-2020-spec-constants.cpp

```diff
@@ -4,7 +4,7 @@
 // RUN: sycl-post-link %t.bc -spec-const=emulation -o %t-split2.txt
 // RUN: cat %t-split2_0.prop | FileCheck %s -check-prefixes=CHECK,CHECK-DEF
 // RUN: llvm-spirv -o %t-split1_0.spv -spirv-max-version=1.1 -spirv-ext=+all %t-split1_0.bc
-// RUN: llvm-spirv -o %t-split2_0.spv -spirv-max-version=1.1 -spirv-ext=+all %t-split2_0.bc
+// RUN: llvm-spirv -o - --to-text %t-split1_0.spv | FileCheck %s -check-prefixes=CHECK-SPV
```

Comment:
Why were we checking the emulation path for SPIR-V targets, which have native support for specialization constants? I may be missing something, but I don't see the value of such test, as specialization constants would be `gep + load` of a kernel argument. Replace with test of generated SPIR-V for native support.

Reply:
> Why were we checking the translation to SPIR-V for the emulation path when SPIR-V has native support for this?

It was 3 years ago and I don't remember already :)

The comment below says:

> The test checks that the tool chain correctly identifies all specialization
> constants, emits correct specialization constats map file and can properly
> translate the resulting bitcode to SPIR-V.

I guess my intent was to have some kind of integration test which would also check that whatever we emit for the emulation path can be properly translated into SPIR-V. It is probably not that bad to have it, but at the same time I don't see any special value in it, so I have no objections to remove it.

Reply:
I'd argue we can remove, but I don't have a super strong opinion. Do rest of changes LGTY?